### ISSUE TITLE
revision is no longer a key of label...

### ISSUE THREAD

**Issue by adamyi** [kiwiz/gkeepapi#24]
Log in an account with a label
```
  File "/Library/Python/2.7/site-packages/gkeepapi/__init__.py", line 348, in login
    self.load(auth)
  File "/Library/Python/2.7/site-packages/gkeepapi/__init__.py", line 362, in load
    self.sync()
  File "/Library/Python/2.7/site-packages/gkeepapi/__init__.py", line 566, in sync
    self._parseUserInfo(changes['userInfo'])
  File "/Library/Python/2.7/site-packages/gkeepapi/__init__.py", line 629, in _parseUserInfo
    node.load(label)
  File "/Library/Python/2.7/site-packages/gkeepapi/node.py", line 1353, in load
    self._revision = raw['revision']
KeyError: 'revision'
```

FYI, userInfo looks like this now:
```
{u'timestamps': {u'kind': u'notes#timestamps'}, u'contextualCoachmarksAcked': [u'keep_it_cc'], u'labels': [{u'timestamps': {u'updated': u'2018-04-27T09:37:43.175Z', u'created': u'2018-04-27T09:37:42.417Z'}, u'name': u'test', u'mainId': u'tag.0yhepizup3b1.16306772811'}], u'settings': {u'singleSettings': [{u'applicablePlatforms': [u'ANDROID', u'WEB', u'CRX', u'IOS', u'CRX_BA'], u'type': u'WEB_EMBEDS_ENABLED', u'webEmbedsEnabledValue': True}, {u'globalCheckedListItemsPolicyValue': u'GRAVEYARD', u'applicablePlatforms': [u'ANDROID', u'WEB', u'CRX', u'IOS', u'CRX_BA'], u'type': u'GLOBAL_CHECKED_LIST_ITEMS_POLICY'}, {u'applicablePlatforms': [u'ANDROID', u'WEB', u'CRX', u'IOS', u'CRX_BA'], u'globalNewListItemPlacementValue': u'BOTTOM', u'type': u'GLOBAL_NEW_LIST_ITEM_PLACEMENT'}, {u'applicablePlatforms': [u'ANDROID', u'WEB', u'CRX', u'IOS', u'CRX_BA'], u'type': u'SHARING_ENABLED', u'sharingEnabledValue': True}, {u'layoutStyleValue': u'GRID', u'applicablePlatforms': [u'ANDROID', u'WEB', u'CRX', u'IOS', u'CRX_BA'], u'type': u'LAYOUT_STYLE'}]}}
```

**Comment by hauckwill** [comment]
Getting a similar error, not sure if it's related:

```
Traceback (most recent call last):
  File "index.py", line 6, in <module>
    success = keep.login(USERNAME, PASSWORD)
  File "C:\Python27\lib\site-packages\gkeepapi\__init__.py", line 348, in login
    self.load(auth)
  File "C:\Python27\lib\site-packages\gkeepapi\__init__.py", line 362, in load
    self.sync()
  File "C:\Python27\lib\site-packages\gkeepapi\__init__.py", line 568, in sync
    self._parseNodes(changes['nodes'])
  File "C:\Python27\lib\site-packages\gkeepapi\__init__.py", line 592, in _parseNodes
    node = _node.from_json(raw_node)
  File "C:\Python27\lib\site-packages\gkeepapi\node.py", line 1428, in from_json
    ncls = _type_map[NodeType(_type)]
KeyError: <gkeepapi.node.NodeType object at 0x0000000006BC5F60>
```

**Comment by kiwiz** [comment]
@adamyi Afaik, all the labels in my account have a `revision` key. Have you noticed any other behaviour that differs?
@hauckwill Can you test with `master`? I'm curious what `_type` is set to.

**Comment by hauckwill** [comment]
@kiwiz Tested with master and getting no notes through all(). I am getting a new message: No handlers could be found for logger "gkeepapi.node"

I'd be more than happy to help you with whatever you need to diagnose this.

**Comment by kiwiz** [comment]
Try adding `logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)` at
the top of your script. Can you attach the resulting output (feel free to
anonymize the IDs but please keep references to `root`)?

**Comment by hauckwill** [comment]
Got it.

For reference, this account has two shared notes (that are not owned by the account) and everything was working fine a couple of days ago on the pip version.

```
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): android.clients.google.com
DEBUG:urllib3.connectionpool:https://android.clients.google.com:443 "POST /auth HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): android.clients.google.com
DEBUG:urllib3.connectionpool:https://android.clients.google.com:443 "POST /auth HTTP/1.1" 200 None
DEBUG:keep:Starting sync: None
DEBUG:keep:Syncing 0 labels and 0 nodes
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): www.googleapis.com
DEBUG:urllib3.connectionpool:https://www.googleapis.com:443 "POST /notes/v1/changes HTTP/1.1" 200 None
WARNING:gkeepapi.node:Unknown node type: LIST
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: NOTE
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
WARNING:gkeepapi.node:Unknown node type: LIST_ITEM
DEBUG:keep:Discarded unknown node
DEBUG:keep:Finishing sync: CgQ0NjYzEhBDZ2tJN0xpZjhacnMyZ0k9
DEBUG:keep:Starting sync: CgQ0NjYzEhBDZ2tJN0xpZjhacnMyZ0k9
DEBUG:keep:Syncing 0 labels and 0 nodes
DEBUG:urllib3.connectionpool:https://www.googleapis.com:443 "POST /notes/v1/changes HTTP/1.1" 200 None
DEBUG:keep:Finishing sync: CgQ0NjYzEhBDZ2tJN0xpZjhacnMyZ0k9
```

**Comment by kiwiz** [comment]
Some quick checks:

- What version of Python are you running?
- Do you have the `enum` or `enum34` module installed (
https://stackoverflow.com/questions/36990882/how-to-live-with-both-enum-and-enum34
)?

**Comment by hauckwill** [comment]
Python 2.7.14

I had enum installed, and after adding enum34 it worked flawlessly!

Thank you so much for the help. Really appreciate the work you've done on this!

**Comment by m-wrzr** [comment]
I have the same issue in python 3.6. Uncommenting ```self._revision``` allows login, however there are still many sync attempts (~20). Is this normal?

**Comment by kiwiz** [comment]
@m-wrzr Do you have a large number of notes or does it look like it's churning on the same few notes? The former is perfectly normal!

**Comment by m-wrzr** [comment]
The nodes are creates at the beginning, afterwards it repeats syncing. How can I provide more debug info?

```DEBUG:urllib3.connectionpool:https://www.googleapis.com:443 "POST /notes/v1/changes HTTP/1.1" 200 None
DEBUG:keep:Finishing sync: CqcCYTQ4MjM0LkVBSWF4Z0VMRWdCU3ZRRUtxd0VLZnZqME8tYl9fX19fd3hXV0pTRFF3VUFBX3dIX19rTnZjMjF2TG5WelpYSW9NREF3TURBd05URTVNVGhoWWpRMFlWVXBMbVJwY2w5bGJuUnllU2d6TlRBek16UXhNemMwTVRoY0xqUTFYQzVPTXpVd016TTBNVE0zTkRFNFhDNHhOakZoWWpnNE9EUm1ZbHd1T0RGaU1UVmlaRE0wTlRZeU5EUXpOU2tBQVJCNEllV3N4QTBQWmZFaU9RQUFBQUFaeEFzQVNBUlFBRm9MQ1JLSTY0R3FfWkM1RUFKZzNiU3o1UVlnQVRJR0NKam9zZGdGOElTVmZBRU1RQUFpQ3dsS3RJcVJVUUFBQUNBRxIQQ2drSXFlRHlzcXlwMndJPQ==
DEBUG:keep:Starting sync: CqcCYTQ4MjM0LkVBSWF4Z0VMRWdCU3ZRRUtxd0VLZnZqME8tYl9fX19fd3hXV0pTRFF3VUFBX3dIX19rTnZjMjF2TG5WelpYSW9NREF3TURBd05URTVNVGhoWWpRMFlWVXBMbVJwY2w5bGJuUnllU2d6TlRBek16UXhNemMwTVRoY0xqUTFYQzVPTXpVd016TTBNVE0zTkRFNFhDNHhOakZoWWpnNE9EUm1ZbHd1T0RGaU1UVmlaRE0wTlRZeU5EUXpOU2tBQVJCNEllV3N4QTBQWmZFaU9RQUFBQUFaeEFzQVNBUlFBRm9MQ1JLSTY0R3FfWkM1RUFKZzNiU3o1UVlnQVRJR0NKam9zZGdGOElTVmZBRU1RQUFpQ3dsS3RJcVJVUUFBQUNBRxIQQ2drSXFlRHlzcXlwMndJPQ==
DEBUG:keep:Syncing 0 labels and 0 nodes
DEBUG:urllib3.connectionpool:https://www.googleapis.com:443 "POST /notes/v1/changes HTTP/1.1" 200 None
DEBUG:keep:Finishing sync: Cq4CYTQ4MjM0LkVBSWF5d0VMRWdCU3dnRUtzQUVLZ1FINDlEWnRfX19fXzhNVmc4MVNZSGdBX184QUFQOEJfXzVEYjNOdGJ5NTFjMlZ5S0RBd01EQXdNRFV4T1RFNFlXSTBOR0ZWS1M1a2FYSmZaVzUwY25rb016VXdNek0wTVRNM05ERTRYQzQwTlZ3dVRqTTFNRE16TkRFek56UXhPRnd1TVRZd04yUTJZemMyTjJaY0xqbGhNRGcwTldZd04yTTFZamsyWmpRcEFBRVE2QUlocmRwUWtLUl9NY3M1QUFBQUFKTEpDd0JJQkZBQVdnc0pFb2pyZ2FyOWtMa1FBbURkdExQbEJpQUJNZ1lJbU9peDJBWHdoSlY4QVF4QUFDSUxDVXEwaXBGUkFBQUFJQVkSEENna0lxZUR5c3F5cDJ3ST0=
DEBUG:keep:Starting sync: Cq4CYTQ4MjM0LkVBSWF5d0VMRWdCU3dnRUtzQUVLZ1FINDlEWnRfX19fXzhNVmc4MVNZSGdBX184QUFQOEJfXzVEYjNOdGJ5NTFjMlZ5S0RBd01EQXdNRFV4T1RFNFlXSTBOR0ZWS1M1a2FYSmZaVzUwY25rb016VXdNek0wTVRNM05ERTRYQzQwTlZ3dVRqTTFNRE16TkRFek56UXhPRnd1TVRZd04yUTJZemMyTjJaY0xqbGhNRGcwTldZd04yTTFZamsyWmpRcEFBRVE2QUlocmRwUWtLUl9NY3M1QUFBQUFKTEpDd0JJQkZBQVdnc0pFb2pyZ2FyOWtMa1FBbURkdExQbEJpQUJNZ1lJbU9peDJBWHdoSlY4QVF4QUFDSUxDVXEwaXBGUkFBQUFJQVkSEENna0lxZUR5c3F5cDJ3ST0=
DEBUG:keep:Syncing 0 labels and 0 nodes
DEBUG:urllib3.connectionpool:https://www.googleapis.com:443 "POST /notes/v1/changes HTTP/1.1" 200 None
DEBUG:keep:Finishing sync: CqkCYTQ4MjM0LkVBSWF4d0VMRWdCU3ZnRUtyQUVLZnZqME1SVF9fX19fd3hWeDNxemhXaUFBX3dIX19rTnZjMjF2TG5WelpYSW9NREF3TURBd05URTVNVGhoWWpRMFlWVXBMbVJwY2w5bGJuUnllU2d6TlRBek16UXhNemMwTVRoY0xqUTFYQzVPTXpVd016TTBNVE0zTkRFNFhDNHhOV1kxWVRBMlpEWmlabHd1WWpGak56Y3dZVEE1TVRWaFpEVmlaQ2tBQVJEWUJDSGxyTVFORDJYeElqa0FBQUFBNjg0TEFFZ0NVQUJhQ3drU2lPdUJxdjJRdVJBQ1lOMjBzLVVHSUFFeUJnaVk2TEhZQmZDRWxYd0JERUFBSWdzSlNyU0trVkVBQUFBZ0JnEhBDZ2tJcWVEeXNxeXAyd0k9
DEBUG:keep:Starting sync: CqkCYTQ4MjM0LkVBSWF4d0VMRWdCU3ZnRUtyQUVLZnZqME1SVF9fX19fd3hWeDNxemhXaUFBX3dIX19rTnZjMjF2TG5WelpYSW9NREF3TURBd05URTVNVGhoWWpRMFlWVXBMbVJwY2w5bGJuUnllU2d6TlRBek16UXhNemMwTVRoY0xqUTFYQzVPTXpVd016TTBNVE0zTkRFNFhDNHhOV1kxWVRBMlpEWmlabHd1WWpGak56Y3dZVEE1TVRWaFpEVmlaQ2tBQVJEWUJDSGxyTVFORDJYeElqa0FBQUFBNjg0TEFFZ0NVQUJhQ3drU2lPdUJxdjJRdVJBQ1lOMjBzLVVHSUFFeUJnaVk2TEhZQmZDRWxYd0JERUFBSWdzSlNyU0trVkVBQUFBZ0JnEhBDZ2tJcWVEeXNxeXAyd0k9
DEBUG:keep:Syncing 0 labels and 0 nodes
DEBUG:urllib3.connectionpool:https://www.googleapis.com:443 "POST /notes/v1/changes HTTP/1.1" 200 None
DEBUG:keep:Finishing sync: CqkCYTQ4MjM0LkVBSWF4d0VMRWdCU3ZnRUtyQUVLZnZqMEs0Yl9fX19fd3hWZlBzakRLTUFBX3dIX19rTnZjMjF2TG5WelpYSW9NREF3TURBd05URTVNVGhoWWpRMFlWVXBMbVJwY2w5bGJuUnllU2d6TlRBek16UXhNemMwTVRoY0xqUTFYQzVPTXpVd016TTBNVE0zTkRFNFhDNHhOV1V5TjJJd056STBNRnd1WWpZMU56QmxNV1ptTXpKa1pUWmpNQ2tBQVJESUJpR3QybENRcEg4eHl6a0FBQUFBZWRRTEFFZ0NVQUJhQ3drU2lPdUJxdjJRdVJBQ1lOMjBzLVVHSUFFeUJnaVk2TEhZQmZDRWxYd0JERUFBSWdzSlNyU0trVkVBQUFBZ0JnEhBDZ2tJcWVEeXNxeXAyd0k9
DEBUG:keep:Starting sync: CqkCYTQ4MjM0LkVBSWF4d0VMRWdCU3ZnRUtyQUVLZnZqMEs0Yl9fX19fd3hWZlBzakRLTUFBX3dIX19rTnZjMjF2TG5WelpYSW9NREF3TURBd05URTVNVGhoWWpRMFlWVXBMbVJwY2w5bGJuUnllU2d6TlRBek16UXhNemMwTVRoY0xqUTFYQzVPTXpVd016TTBNVE0zTkRFNFhDNHhOV1V5TjJJd056STBNRnd1WWpZMU56QmxNV1ptTXpKa1pUWmpNQ2tBQVJESUJpR3QybENRcEg4eHl6a0FBQUFBZWRRTEFFZ0NVQUJhQ3drU2lPdUJxdjJRdVJBQ1lOMjBzLVVHSUFFeUJnaVk2TEhZQmZDRWxYd0JERUFBSWdzSlNyU0trVkVBQUFBZ0JnEhBDZ2tJcWVEeXNxeXAyd0k9
DEBUG:keep:Syncing 0 labels and 0 nodes
```
and so on

**Comment by kiwiz** [comment]
@m-wrzr should be fixed in 0.10.7.

**Comment by m-wrzr** [comment]
Works great, thanks!
